packages/react-component-library/src/components/Alert/Alert.test.tsx

```diff
@@ -203,4 +203,24 @@ describe('Alert', () => {
       )
     })
   })
+
+  describe('when hideDismiss prop is provided', () => {
+    beforeEach(() => {
+      wrapper = render(<Alert hideDismiss>Description</Alert>)
+    })
+
+    it('should not render the close button', () => {
+      expect(wrapper.queryByTestId('close')).not.toBeInTheDocument()
+    })
+
+    it('should still render the alert content', () => {
+      expect(wrapper.getByTestId('content-description')).toHaveTextContent(
+        'Description'
+      )
+    })
+
+    it('should render the state icon', () => {
+      expect(wrapper.getByTestId('state-icon')).toBeInTheDocument()
```

Comment:
I think this is fine as it's consistent with other testing but I might be tempted to start using `getByRole('button', { name: 'Dismiss' })`